h5py_wrapper/lib.py

```diff
@@ -56,3 +57,11 @@ def convert_numpy_types_in_dict(d):
             d[key] = float(value)
         elif isinstance(value, (np.bool_)):
             d[key] = bool(value)
+
+
+def convert_iterable_to_numpy_array(it):
+    array = np.array(it)
```

Comment:
This function needs some explanation, such as:
"""
Converts an iterable to a numpy array. If the elements of the iterable are strings, they are converted to `np.string_`.
"""